### PR TITLE
spack recipes for rccl rocprim rocthrust

### PR DESCRIPTION
**Pull request by gkathirv**
New recipes

### CODE REVIEW

File: packages/rccl/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from spack import *
```
Comment (arjun-raj-kuppala) [inline packages/rccl/package.py:6]: Need to add common license which i know under discussion.

File: packages/rocthrust/package.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        if spec.variants['clients'].value == True:
+            args += [
+                     '-DBUILD_BENCHMARK=ON',
+                     '-DBUILD_TEST=ON',
+                    ]
+
+        return args
+
+    @run_after('install')
```
Comment (arjun-raj-kuppala) [inline packages/rocthrust/package.py]: As per the math libs team(Eiden), clients testing is need for internal testing not for public, we may need to remove this

File: packages/rocprim/package.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        if spec.variants['clients'].value == True:
+            args += [
+                     '-DBUILD_BENCHMARK=ON',
+                     '-DBUILD_TEST=ON',
+                    ]
+
+        return args
+
+    @run_after('install')
+    def move_clients(self):
```
Comment (arjun-raj-kuppala) [inline packages/rocprim/package.py]: As per the math libs team(Eiden), clients testing is need for internal testing not for public, we may need to remove this

File: packages/rocprim/package.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on('hsa-rocr-dev@3.5.0:', type='build', when='@3.5:')
+
+    def cmake_args(self):
+        spec=self.spec
+
+        args = [
+                '-DCMAKE_CXX_COMPILER={}/hipcc'.format(spec['hip'].prefix.bin),
+                '-DUSE_HIP_CLANG=ON',
+                '-DCMAKE_MODULE_PATH={}/cmake'.format(spec['hip'].prefix),
+                '-DHIP_CLANG_INCLUDE_PATH={}/lib/clang/11.0.0/include'.format(self.spec['llvm-amdgpu'].prefix)
```
Comment (arjun-raj-kuppala) [inline packages/rocprim/package.py]: Version number of clang should be got dynamically, i have updated snippet to get the same for rocrand, we can use the same going forward.

File: packages/hipcub/package.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    depends_on('hsa-rocr-dev@3.5.0:', type='build', when='@3.5:')
+
+    def cmake_args(self):
+        spec=self.spec
+
+        args = [
+                '-DCMAKE_CXX_COMPILER={}/hipcc'.format(spec['hip'].prefix.bin),
+                '-DUSE_HIP_CLANG=ON',
+                '-DCMAKE_MODULE_PATH={}/cmake'.format(spec['hip'].prefix),
+                '-DHIP_CLANG_INCLUDE_PATH={}/lib/clang/11.0.0/include'.format(self.spec['llvm-amdgpu'].prefix)
```
Comment (arjun-raj-kuppala) [inline packages/hipcub/package.py:33]: You can add the dynamic version extraction snippet here

**Comment by gkathirv** [comment]
Addressed the review comments, please review again

**Review by streamhsa — APPROVED** [review]

**Pull request merged**